src/components/EditProfile/index.jsx

```diff
@@ -78,65 +78,124 @@ const EditProfile = ({ userData, username, setIsEditing, setUserData }) => {
     }
   };
 
-  const handleImgSave = () => {
-    if (!usernameAvailable) {
-      return;
-    }
-    const oldImg = userData.avatar;
-    if (image) {
-      const uploadTask = storage.ref(`images/${image?.name}`).put(image);
-      uploadTask.on(
-        "state_changed",
-        () => {},
-        (error) => {
-          // playErrorSound();
-          enqueueSnackbar(error.message, {
-            variant: "error",
-          });
-        },
-        () => {
-          storage
-            .ref("images")
-            .child(image?.name)
-            .getDownloadURL()
-            .then(async (url) => {
-              //Updating profile data in auth
-              await auth.currentUser.updateProfile({
-                displayName: name,
-                photoURL: url,
-              });
+  const [error, setError] = useState({});
+  const handleImgSave = (event) => {
+    event.preventDefault();
+    console.log("called");
+    console.log(editedData.name);
+    if (editedData.name === "") {
+      setError("**Name is Required!");
+    } else if (editedData.newUsername === "") {
+      setError("**User name is Required!");
+    } else if (editedData.country === "") {
+      setError("**Country is Required!");
+    } else if (editedData.bio === "") {
+      setError("**Bio is Required!");
+    } else {
```

Comment:
also, can we pick the above logic in a separate function to improve readibility?